homeassistant/components/hunterdouglas_powerview/__init__.py

```diff
@@ -126,3 +127,44 @@ async def async_get_device_info(hub: Hub) -> PowerviewDeviceInfo:
 async def async_unload_entry(hass: HomeAssistant, entry: PowerviewConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: PowerviewConfigEntry) -> bool:
+    """Migrate entry."""
+
+    _LOGGER.debug("Migrating from version %s.%s", entry.version, entry.minor_version)
+
+    if entry.version == 1:
+        # 1 -> 2: Unique ID from integer to string
+        if entry.minor_version == 1:
+            await _migrate_unique_ids(hass, entry)
+            hass.config_entries.async_update_entry(entry, minor_version=2)
+
+    _LOGGER.debug("Migrated to version %s.%s", entry.version, entry.minor_version)
+
+    return True
+
+
+async def _migrate_unique_ids(hass: HomeAssistant, entry: PowerviewConfigEntry) -> None:
+    """Migrate int based unique ids to str."""
+    entity_registry = er.async_get(hass)
+    registry_entries = er.async_entries_for_config_entry(
+        entity_registry, entry.entry_id
+    )
+    for reg_entry in registry_entries:
+        if isinstance(reg_entry.unique_id, int) or (
+            isinstance(reg_entry.unique_id, str)
+            and entry.unique_id is not None
+            and not reg_entry.unique_id.startswith(entry.unique_id)
+        ):
+            _LOGGER.debug(
+                "Migrating %s: %s to %s_%s",
+                reg_entry.entity_id,
+                reg_entry.unique_id,
+                entry.unique_id,
+                reg_entry.unique_id,
+            )
+            entity_registry.async_update_entity(
+                reg_entry.entity_id,
+                new_unique_id=f"{entry.unique_id}_{reg_entry.unique_id}",
+            )
```

Comment:
We really want to have tests for this since changing unique_ids is delicate and when that goes wrong its an even bigger mess.

Reply:
yea i saw this one was not covered by codecov and was afraid you would say that 🙈 
Bit of a noob with tests - can you point me in the direction of an integration that has tests for similar ?